apps/settings/src/components/AppAPI/DaemonSelectionDialog.vue

```diff
@@ -0,0 +1,63 @@
+<!--
+  - SPDX-FileCopyrightText: 2025 Nextcloud GmbH and Nextcloud contributors
+  - SPDX-License-Identifier: AGPL-3.0-or-later
+-->
+<template>
+	<NcDialog :open="show"
+		:name="t('settings', 'Choose Deploy Daemon for {appName}', {appName: app.name })"
+		size="normal"
+		@update:open="closeModal">
+		<DaemonSelectionList :app="app"
+			:deploy-options="deployOptions"
+			@close="closeModal" />
+	</NcDialog>
+</template>
+
+<script>
+import NcDialog from '@nextcloud/vue/components/NcDialog'
+import DaemonSelectionList from './DaemonSelectionList.vue'
+import { useAppsStore } from '../../store/apps-store'
+import { useAppApiStore } from '../../store/app-api-store'
+
+export default {
+	name: 'DaemonSelectionDialog',
+	components: {
+		NcDialog,
+		DaemonSelectionList,
+	},
+	props: {
+		show: {
+			type: Boolean,
+			required: true,
+		},
+		app: {
+			type: Object,
+			required: true,
+		},
+		deployOptions: {
+			type: Object,
+			required: false,
+			default: () => ({}),
+		},
+	},
+	setup() {
+		const store = useAppsStore()
+		const appApiStore = useAppApiStore()
+
+		return {
+			store,
+			appApiStore,
+		}
+	},
+	data() {
+		return {
+			selectDaemonModal: false,
+		}
+	},
+	methods: {
+		closeModal() {
+			this.$emit('update:show', false)
+		},
+	},
+}
+</script>
```

Comment:
For new components I would recommend - in the future to directly use script-setup. That will make life much easier when we migrate to vue 3.

E.g. this could look like this:
```suggestion
<template>
	<NcDialog :open="show"
		:name="t('settings', 'Choose Deploy Daemon for {appName}', {appName: app.name })"
		size="normal"
		@update:open="closeModal">
		<DaemonSelectionList :app="app"
			:deploy-options="deployOptions"
			@close="closeModal" />
	</NcDialog>
</template>

<script setup>
import { ref } from 'vue'
import NcDialog from '@nextcloud/vue/components/NcDialog'
import DaemonSelectionList from './DaemonSelectionList.vue'
import { useAppsStore } from '../../store/apps-store'
import { useAppApiStore } from '../../store/app-api-store'

const props = defineProps({
	show: {
		type: Boolean,
		required: true,
	},
	app: {
		type: Object,
		required: true,
	},
	deployOptions: {
		type: Object,
		default: () => ({}),
	},
})

const emit = defineEmits(['update:show'])

const store = useAppsStore()
const appApiStore = useAppApiStore()

const selectDaemonModal = ref(false)

function closeModal() {
	emit('update:show', false)
}
</script>
```